metricshub-agent/src/main/java/org/sentrysoftware/metricshub/agent/helper/ConnectorVariablesLibraryParser.java

```diff
@@ -0,0 +1,288 @@
+			if (Files.isDirectory(path) || !isYamlFile(path.toFile().getName())) {
+				return FileVisitResult.CONTINUE;
+			}
+			final ObjectMapper yamlMapper = new YAMLMapper();
+			final JsonNode connectorNode = yamlMapper.readTree(path.toFile());
+			if (!isConnector(connectorNode)) {
+				return FileVisitResult.CONTINUE;
+			}
+
+			// Get the connector's file name
+			final String filename = path.getFileName().toString();
+			final String connectorId = filename.substring(0, filename.lastIndexOf('.'));
+
+			if (!connectorNode.toString().contains("${var::")) {
+				return FileVisitResult.CONTINUE;
+			}
+
+			// Normalize additionalConnectors
+			normalizeAdditionalConnectors();
+
+			// Removing all the configurations that are not using this connector.
+			final Map<String, AdditionalConnector> filteredConnectors = additionalConnectorConfig
+				.entrySet()
+				.stream()
+				.filter(entry -> connectorId.equalsIgnoreCase(entry.getValue().getUses()))
+				.collect(
+					Collectors.toMap(
+						Map.Entry::getKey, // Keep the original key
+						Map.Entry::getValue // Keep the original AdditionalConnectorConfig as the value
+					)
+				);
+
+			// Construct a variables map from the default connector variables.
+			final Map<String, String> defaultVariables = new HashMap<>(getDefaultConnectorVariables(connectorNode));
+
+			// Parse the connector even if it is not configured as an additional connector.
+			// This ensures that the connector will function with the default variables if the user forces it.
+			if (filteredConnectors.isEmpty()) {
+				parseConnectorWithModifier(path, defaultVariables, connectorId, filename, connector -> {});
+			}
+
+			// For each configuration, we create a new custom connector and a new variables map to be used in the connector update.
+			for (final Entry<String, AdditionalConnector> connectorConfigurationEntry : filteredConnectors.entrySet()) {
+				final String additionalConnectorId = connectorConfigurationEntry.getKey();
+				final AdditionalConnector additionalConnectorValue = connectorConfigurationEntry.getValue();
+
+				// Add the connector to the host connectors set.
+				connectorsParsingResult
+					.getHostConnectors()
+					.add(additionalConnectorValue.isForce() ? "+" + additionalConnectorId : additionalConnectorId);
+
+				// Retrieve and use default connector variables on this connector for this configuration.
+				final Map<String, String> connectorVariables = new HashMap<>(defaultVariables);
+
+				// Override the default connector variables by the connector variables that the user configured.
+				final Map<String, String> configuredVariables = additionalConnectorValue.getVariables();
+				if (configuredVariables != null) {
+					connectorVariables.putAll(configuredVariables);
+				}
+				// There are at least two additional connectors that use the current connector.
+				// This means that the compiled filename of these connectors needs to be modified.
+				if (filteredConnectors.size() > 1) {
```

Comment:
We should always proceed regardless of the size of the connector configurations that define the `uses` property, as there's a normalizer (normalizeAdditionalConnectors();) that runs before this loop. I would handle this uniformly, whether the size is 1 or greater than 1.
```java
parseConnectorWithModifier(
	path,
	connectorVariables,
	additionalConnectorId,
	filename,
        connector -> connector.getConnectorIdentity().setCompiledFilename(additionalConnectorId)
);
```
Moreover the user expects to see `additionalConnectorId` instead of the original connect ID even if he has configured only one.